src/basic/log.c

```diff
@@ -32,12 +32,14 @@
 #include "stdio-util.h"
 #include "string-table.h"
 #include "string-util.h"
+#include "strv.h"
 #include "syslog-util.h"
 #include "terminal-util.h"
 #include "time-util.h"
 #include "utf8.h"
 
 #define SNDBUF_SIZE (8*1024*1024)
+#define MAX_IOVEC 128
```

Comment:
usually we put the MAX last, i.e. IOVEC_MAX?

Reply:
btw, why do this? why not instead a counter also as thread_local variable that encodes how many entries are needed? it's increased whenever a context is added, and decreased whenever one is removed.

Reply:
To avoid unbounded stack allocations, I figure it makes sense to impose an upper limit on the amount of stuff we put into the iovec. If we already have a max size, tracking the size of the log context linked list individually doesn't seem like it provides much benefit